templates/taxi/car_confirm_delete.html

```diff
@@ -1,10 +1,18 @@
 {% extends "base.html" %}
 
 {% block content %}
-  <h1>Delete car?</h1>
-  <form action="" method="post">
+  <h3>Are you sure, you want to delete car "{{ car.model }}"?</h3>
+  <<form action="" method="post">
     {% csrf_token %}
-
+    
     <input type="submit" value="Yes" class="btn btn-danger">
-  </form>
-{% endblock %}
+
+    {% if messages %}
+        {% for message in messages %}
+            <a href="{{ message }}" class="btn btn-secondary">Cancel</a>
+        {% endfor %}
+    {% else %}
+        <a href="{% url 'taxi:car-list' %}" class="btn btn-secondary">Cancel</a>
+    {% endif %}
```

Comment:
The `Cancel` button should lead to the previous page the user was on. It's not clear how the `messages` variable is being used here. The `Cancel` button should not depend on the presence of messages. Also, it's not a good practice to use the `messages` framework for this purpose. It's better to use the `HTTP_REFERER` header to get the URL of the previous page.